File: src/containers/missions.ts
import type { Mission } from "@prisma/client";
import { db } from "../db/client";

export class MissionContainer {
  private missions: Mission[] = [];

  async bootstrap(): Promise<void> {
    await this.reload();
  }

  async reload(): Promise<void> {
    this.missions = await db.mission.findMany({
      orderBy: {
        name: "asc",
      },
    });
  }

  getMissions(): Mission[] {
    return this.missions;
  }

  getMissionWithName(name: string): Mission | null {
    const finalName = name
      .toUpperCase()
      .replace(/\s/g, "")
      .replace("HARD", "H");
    return this.missions.find((mission) => mission.name === finalName) || null;
  }
}

export const missionContainer = new MissionContainer();

File: src/containers/missions.rs
use crate::db::models::Mission;

#[derive(Debug, Default)]
pub struct MissionContainer {
    missions: Vec<Mission>,
}

impl MissionContainer {
    pub fn new() -> Self {
        Self::default()
    }

    pub fn reload(&mut self, missions: Vec<Mission>) {
        self.missions = missions;
    }

    pub fn get_missions(&self) -> &[Mission] {
        &self.missions
    }

    pub fn get_mission_with_name(&self, name: &str) -> Option<&Mission> {
        let final_name = name
            .to_uppercase()
            .replace(char::is_whitespace, "")
            .replace("HARD", "H");
        self.missions.iter().find(|m| m.name == final_name)
    }
}
